src/Raven.Client/Documents/Session/AsyncDocumentSession.cs

```diff
@@ -126,6 +126,9 @@ protected override Task<string> GenerateIdAsync(object entity)
         public IAttachmentsSessionOperationsAsync Attachments => _attachments ?? (_attachments = new DocumentSessionAttachmentsAsync(this));
         private IAttachmentsSessionOperationsAsync _attachments;
 
+        public IRetiredAttachmentsSessionOperationsAsync RetiredAttachments => _retiredAttachments ??= new DocumentSessionRetiredAttachmentsAsync(this);
```

Comment:
maybe put that inside Attachments, so the usage will be session.Attachments.Retired ?